src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -93,11 +102,50 @@ public void removePerson(Person key) {
         persons.remove(key);
     }
 
+    //// tutorial-level operations
+
+    /**
+     * Returns true if a person with the same identity as {@code person} exists in the address book.
+     */
+    public boolean hasTutorial(Tutorial tutorial) {
+        requireNonNull(tutorial);
+        return tutorials.contains(tutorial);
+    }
+
+    /**
+     * Adds a tutorial to the address book.
+     * The tutorial must not already exist in the address book.
```

Comment:
Not sure if it is mandatory for javadocs to contain @param if there are params specified in the function. In this case it is very clear what Tutorial p is, but under https://se-education.org/guides/conventions/java/intermediate.html#comments it does not seem to say that @param can be excluded.. maybe we can check with our tutor on this. We can still merge the PR and add these things later if we need to.

Reply:
ok, sounds good to me. Let's check with Bryann in the upcoming tutorial. I shall leave it here and merge it into master for now